instructor/common.py

```diff
@@ -68,22 +68,38 @@ def create_new_assignment(request):
 
 
 def assignment_setup(request):
+    error = ''
     if request.method == "POST":
-        # Have to give error for empty name
-        request.session['assignment_name'] = request.POST['name']
+        assignment_name = request.POST['name']
         if request.POST['based_on']:
             request.session['based_on'] = request.POST['based_on']
-
-        if 'continue' in request.POST:
-            return redirect('common_course_setup')
+        if assignment_name:
+            request.session['assignment_name'] = assignment_name
+            if 'continue' in request.POST:
+                return redirect('common_course_setup')
+        else:  # Have to give error for empty name
+            error = "*Empty assignment name"
+    else:
+        # giving a default name to the assignment
+        assignment_name = "Assignment"
+        filtered_asgmts = models.Assignment.objects.filter(name__regex=r'Assignment.*')
+        # want to allow index one larger then there are currently assignments
+        for index in xrange(1, len(filtered_asgmts)+2):
```

Comment:
because xrange(1,n) gives a range [1, n-1]. If I already have n assignments in the database and in the worst case all of them are named "Assignment "+index then I will need the index to be able to take on the value of n+1.